src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -401,7 +401,7 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
                 diagnostics.Add(ErrorCode.ERR_PointerTypeInPatternMatching, typeSyntax.Location);
                 return true;
             }
-            else if (patternType.IsNullableType() && patternTypeWasInSource)
+            else if (patternType.IsNullableType() || typeSyntax is NullableTypeSyntax && patternTypeWasInSource)
```

Comment:
>patternType.IsNullableType() || typeSyntax is NullableTypeSyntax [](start = 21, length = 64)

Should this expression be parenthesized? (Should `patternTypeWasInSource` apply regardless?)

Reply:
I will remove `patternTypeWasInSource`.  It is always `true`.

---
In reply to: [407620967](https://github.com/dotnet/roslyn/pull/43280#discussion_r407620967) [](ancestors = 407620967)